inginious/agent/docker_agent/_docker_interface.py

```diff
@@ -150,13 +165,14 @@ def create_container(self, image, network_grading, mem_limit, task_path, sockets
                 course_common_student_path: {'bind': '/course/common/student', 'mode': 'ro'}
             },
             runtime=runtime,
-            ulimits=[nofile_limit]
+            ulimits=[nofile_limit],
+            devices=['/dev/kvm'] if kvm and image_requires_kvm else []
```

Comment:
I fear that unwanted behaviours can happen silently here. If the container image requires kvm and the agent does not support/enable it, no notification will be given. There seems to be no way for the user to know if the agent supports kvm, and, moreover, nothing prevents the job to be assigned to an inappropriate agent.

I'm however more likely to deploy an agent that do support kvm into a cluster of other agents that don't.

For me it would probably be more relevant to implement this as the ssh or nvidia filters : announce another agent type that will allow container images tagged with `org.inginious.kvm` to be listed as available environment, so that we can ensure appropriate routing and ensure the user selected the right environment.

Implementation of `nvidia` is here https://github.com/UCL-INGI/INGInious/commit/f842bee563f8eb7bca3957e4f5019c45dbd68e6d

However, I realize that implementing a kvm filter, as it may be compatible with other agent type, would require to duplicate all those one to have a kvm-enabled flavour. 

Maybe it would be useful to rethink the way this is done and add some kind of feature flags in addition to the agent type, so that we can handle `docker` or `nvidia` with `ssh` or `kvm` as supported features or not.